pkg/scheduler/framework/plugins/legacy_registry.go

```diff
@@ -169,16 +171,16 @@ type ConfigProducerArgs struct {
 	InterPodAffinityArgs *config.InterPodAffinityArgs
 }
 
-// ConfigProducer returns the set of plugins and their configuration for a
+// configProducer returns the set of plugins and their configuration for a
 // predicate/priority given the args.
-type ConfigProducer func(args ConfigProducerArgs) (config.Plugins, []config.PluginConfig)
+type configProducer func(ConfigProducerArgs, *config.Plugins, []config.PluginConfig) []config.PluginConfig
```

Comment:
looks strange that we return PluginConfig but not Plugins.

Reply:
Yeah... I thought about that, but this is just internal now, so maybe it's fine?

Reply:
I think it doesn't matter if it is internal or external, I suggest we change it, but wont insist.

Reply:
Given we modified the plugins (pointer) in place, IMO it's not a bad idea to pass-in a pointer to the pluginConfig slice, and hence to modify it in place.

Reply:
I went with the pointer, just for brevity. I left the copy arguments for the exported functions

Reply:
sgtm